public/app/core/components/Login/ChangePassword.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { PureComponent, SyntheticEvent } from 'react';
+import { Tooltip } from '@grafana/ui';
+import appEvents from 'app/core/app_events';
+
+interface Props {
+  onSubmit: (pw: { newPassword: string; confirmNew: string; oldPassword: string }, valid: boolean) => void;
+  onSkip: Function;
+  className?: string;
+  focus?: boolean;
+}
+
+interface State {
+  newPassword: string;
+  confirmNew: string;
+  valid: boolean;
+}
+
+export class ChangePassword extends PureComponent<Props, State> {
+  private userInput: HTMLInputElement;
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      newPassword: '',
+      confirmNew: '',
+      valid: false,
+    };
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (!prevProps.focus && this.props.focus) {
+      this.focus();
+    }
+  }
+
+  focus() {
+    this.userInput.focus();
+  }
+
+  onSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+
+    const { newPassword, confirmNew } = this.state;
+    if (this.state.valid) {
+      this.props.onSubmit({ newPassword, confirmNew, oldPassword: 'admin' }, true);
+    } else {
+      appEvents.on('alert-warning', ['New passwords do not match', '']);
```

Comment:
Don't you want to `emit` the event rather than listening to it?

Reply:
Good point, although  just realized that you really can't reach this piece of code since the button is disabled as long as the passwords don't match.